### PR TITLE
Add fallback link to Twitter media provider

### LINKED ISSUE #3308: Twitter Media Provider: Add fallback link if the embedding fails

### PR DESCRIPTION
**Pull request by TimWolla**
Created a new dedicated JavaScript module, because fitting this additional logic into the mediaProvider.xml is painful.

-----

Closes #3308

### CODE REVIEW

File: wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Message/TwitterEmbed.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	/**
+	 * @exports	WoltLabSuite/Core/Ui/Message/TwitterEmbed
+	 */
+	return {
+		/**
+		 * Embed the tweet identified by the given tweetId into the given container.
+		 * 
+		 * @param {HTMLElement} container
+		 * @param {string} tweetId
+		 * @param {boolean} clearContents Whether to remove existing children of the given container after embedding the tweet.
```
Comment (TimWolla) [inline wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Message/TwitterEmbed.js:25]:
```suggestion
		 * @param {boolean} removeChildren Whether to remove existing children of the given container after embedding the tweet.
```
Somehow the refactoring didn't rename the variable in the comment.

File: wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Message/TwitterEmbed.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		 */
+		embedTweet: function (container, tweetId, removeChildren) {
+			if (removeChildren === undefined) removeChildren = false;
+			
+			return twitterReady.then(function() {
+				return twttr.widgets.createTweet(tweetId, container, {
+					dnt: true,
+					lang: document.documentElement.lang,
+				});
+			}).then(function (tweet) {
```
Comment (Gravatronics) [inline wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Message/TwitterEmbed.js:36]: If the twitter link is invalid, tweet is undefined causing an error in line 41.

File: com.woltlab.wcf/mediaProvider.xml [hunk 3]
```diff
@@ -78,9 +78,8 @@ https?://www.twitch.tv/[a-zA-Z0-9]+/v/(?<ID>[0-9]+)]]></regex>
...
 			<title>Twitter Tweet</title>
 			<regex><![CDATA[https://twitter.com/(?<USERNAME>[0-9a-zA-Z_]+)/status/(?<ID>[0-9]+)]]></regex>
-			<html><![CDATA[<script>window.twttr=function(t,e,r){var n,i=t.getElementsByTagName(e)[0],w=window.twttr||{};return t.getElementById(r)?w:((n=t.createElement(e)).id=r,n.src="https://platform.twitter.com/widgets.js",i.parentNode.insertBefore(n,i),w._e=[],w.ready=function(t){w._e.push(t)},w)}(document,"script","twitter-wjs");</script>
-<div data-wsc-twitter-tweet="{$ID}"></div>
-<script>twttr.ready(function(t){elBySelAll("[data-wsc-twitter-tweet]",void 0,function(t){elData(t,"wsc-twitter-tweet")&&(twttr.widgets.createTweet(elData(t,"wsc-twitter-tweet"),t,{dnt:!0}),elData(t,"wsc-twitter-tweet",""))})});</script>]]></html>
+			<html><![CDATA[<div data-wsc-twitter-tweet="{$ID}"><a href="https://twitter.com/{$USERNAME}/status/{$ID}/">https://twitter.com/{$USERNAME}/status/{$ID}/</a></div>
+<script>require(["WoltLabSuite/Core/Ui/Message/TwitterEmbed"],function(t){elBySelAll("[data-wsc-twitter-tweet]",void 0,function(e){elData(e,"wsc-twitter-tweet")&&t.embedTweet(e,elData(e,"wsc-twitter-tweet"),!0),elData(e,"wsc-twitter-tweet","")})});</script>]]></html>
```
Comment (Gravatronics) [inline com.woltlab.wcf/mediaProvider.xml:82]: Couldn't the `elBySelAll` code be moved into the module making the minified code in the xml file smaller and easier to comprehend?

File: wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Message/TwitterEmbed.js [hunk 4]
```diff
@@ -43,6 +43,19 @@ define(['https://platform.twitter.com/widgets.js'], function(Widgets) {
 				
 				return tweet;
 			});
+		},
+		/**
```
Comment (Gravatronics) [inline wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Message/TwitterEmbed.js:47]:
```suggestion
		
		/**
```

**Pull request merged**